StreetCare/StreetCare/Views/Screens/CommunityEvents/CommunityEventView.swift

```diff
@@ -0,0 +1,193 @@
+                    .frame(height: 50.0)
+                Menu {
+                    ForEach(FilterType.allCases, id: \.self) { filter in
+                        Button(action: {
+                            selectedFilter = filter
+                            applyFilter(filterType: selectedFilter)
+                        }) {
+                            HStack {
+                                Text(filter.rawValue)
+                                Spacer()
+                                // Checkmark if the filter is selected
+                                if selectedFilter == filter {
+                                    Image(systemName: "checkmark")
+                                        .foregroundColor(.blue) // You can change the color if needed
+                                }
+                            }
+                        }
+                    }
+                } label: {
+                    HStack {
+                        ZStack(alignment: .topTrailing) {
+                            // Main Filter icon
+                            Image(systemName: "line.horizontal.3.decrease.circle")
+                                .foregroundColor(.black)
+                            
+                            // Red dot indicator when a filter is active
+                            if selectedFilter != .none {
+                                Circle()
+                                    .fill(Color.red)
+                                    .frame(width: 8, height: 8) // Adjust size as needed
+                                    .offset(x: 10, y: -5) // Position the red dot
+                            }
+                        }
+                        Text("Filter")
+                    }
+                    .padding(.horizontal, 10)
+                    .foregroundColor(.black)
+                }
+            }
+            .overlay(
+                RoundedRectangle(cornerRadius: 15.0)
+                    .stroke(Color.black, lineWidth: 1)
+            )
+            .padding()
+            
+            if viewModel.filteredData.isEmpty {
+                Text(NSLocalizedString(eventType == .future ? "noFutureEventsAvailable" : eventType == .past ? "noPastEventsAvailable" : "noHelpingRequestsAvailable", comment: ""))
+                    .fontWeight(.bold)
+                    .foregroundColor(Color("TextColor"))
+            }
+            
+            if let _ = self.user {
+                if viewModel.filteredData.isEmpty {
+                    Spacer(minLength: 50.0)
+                } else {
+                    List {
+                        ForEach(0..<viewModel.filteredData.count, id: \.self) { index in
+                            if let date = viewModel.filteredData[index].keys.first,
+                               let eventObj = viewModel.filteredData[index][date] {
+                                Section(header: SectionHeaderView(date: date)) {
```

Comment:
Is it possible to add the count of that particular month. Just asking which will also help to show that after filtering the data, how many events are available.

Reply:
We can do a fast follow and resolve in the next deployment